archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1637,50 +1637,119 @@ public boolean apply(JavaClass input) {
+         *
+         * @see #assignableFrom(String)
+         * @see #assignableFrom(DescribedPredicate)
+         * @see #assignableTo(Class)
+         */
         @PublicAPI(usage = ACCESS)
         public static DescribedPredicate<JavaClass> assignableFrom(final Class<?> type) {
             return assignableFrom(type.getName());
         }
 
+        /**
+         * Same as {@link #assignableTo(Class)} but takes a fully qualified class name as an argument instead of a class object.
+         */
         @PublicAPI(usage = ACCESS)
         public static DescribedPredicate<JavaClass> assignableTo(final String typeName) {
             return assignableTo(GET_NAME.is(equalTo(typeName)).as(typeName));
         }
 
+        /**
+         * Same as {@link #assignableFrom(Class)} but takes a fully qualified class name as an argument instead of a class object.
+         */
         @PublicAPI(usage = ACCESS)
         public static DescribedPredicate<JavaClass> assignableFrom(final String typeName) {
             return assignableFrom(GET_NAME.is(equalTo(typeName)).as(typeName));
         }
 
+        /**
+         * Same as {@link #assignableTo(Class)} but returns {@code true} whenever the respective {@link JavaClass}
+         * is assignable to a class that matches the supplied predicate.<br>
+         * This is the opposite of {@link #assignableFrom(DescribedPredicate)}
+         *
+         * @see #assignableTo(Class)
+         * @see #assignableTo(String)
+         * @see #assignableFrom(DescribedPredicate)
+         */
         @PublicAPI(usage = ACCESS)
         public static DescribedPredicate<JavaClass> assignableTo(final DescribedPredicate<? super JavaClass> predicate) {
             return new AssignableToPredicate(predicate);
         }
 
+        /**
+         * Same as {@link #assignableFrom(Class)} but returns {@code true} whenever the respective {@link JavaClass}
+         * is assignable from a class that matches the supplied predicate.<br>
+         * This is the opposite of {@link #assignableTo(DescribedPredicate)}
+         *
+         * @see #assignableFrom(Class)
+         * @see #assignableFrom(String)
+         * @see #assignableTo(DescribedPredicate)
+         */
         @PublicAPI(usage = ACCESS)
         public static DescribedPredicate<JavaClass> assignableFrom(final DescribedPredicate<? super JavaClass> predicate) {
             return new AssignableFromPredicate(predicate);
         }
 
+        /**
+         * @param type the interface type to check for
+         * @return a {@link DescribedPredicate} that returns {@code true}, if the respective {@link JavaClass} implements the supplied
+         *         interface type. I.e. the supplied class must be an interface and the {@link JavaClass} must be a class for this
+         *         to ever be true (it resembles delarations like `class A implements B`, which only works for a class A and
+         *         an interface B)
+         *
```

Comment:
Should we even document
```suggestion
         * @throws InvalidSyntaxUsageException if {@code type} is not an interface
```
?